Documentation/user-guides/scrapeconfig.md

```diff
@@ -0,0 +1,162 @@
+---
+weight: 151
+toc: true
+title: ScrapeConfig CRD
+menu:
+    docs:
+        parent: user-guides
+lead: ""
+images: []
+draft: true
+description: Guide to use ScrapeConfig to scrape targets external to the Kubernetes cluster
+---
+
+Starting with prometheus-operator v0.65.x, one can use the `ScrapeConfig` CRD to scrape targets external to the
+Kubernetes cluster or create scrape configurations that are not possible with the higher level
+`ServiceMonitor`/`Probe`/`PodMonitor` resources.
+
+# Prerequisites
+* `prometheus-operator` `>v0.65.1`
+* `ScrapeConfig` installed in the cluster. Make sure to restart the operator after having pushed the CRD.
+
+## Configure Prometheus or PrometheusAgent to select ScrapeConfigs
+
+Both the Prometheus and PrometheusAgent CRD have a `scrapeConfigSelector` field. This field needs to be set to a list of
+labels to match `ScrapeConfigs`:
+
+```yaml
+spec:
+  scrapeConfigSelector:
+    prometheus: system-monitoring-prometheus
+```
+
+With this example, all `ScrapeConfig` having the `prometheus` label set to `system-monitoring-prometheus` will be used
+to generate scrape configurations.
+
+# Use ScrapeConfig to scrape an external target
+
+`ScrapeConfig` currently supports a limited set of service discoveries:
+* `static_config`
+* `file_sd`
+* `http_sd`
+* `kubernetes_sd`
```

Comment:
Do we plan to document all fields added to ScrapeConfig?

I'm a bit afraid that we could easily miss documentation as time goes on. Providing examples for all service discovery options might make this page quite big 🤔

Reply:
I was actually thinking about that and wanted to discuss having some use case based documentation rather than this (e.g. how to scrape exporters on the kubernetes nodes or how to scrape an external target etc). That way we don't have to document everything in one page but can focus on use cases that drove the implementation.

What do you think?

Reply:
Since we don't have any documentation and people are already asking for guidance on ScrapeConfig, we could merge while the CRD is still small. We could rethink and rewrite it later.


I was raising this question mostly because I was concerned that we would have to have this discussion at some point 🤔

Reply:
I'd be +1 on having the documentation around specific use cases. And I feel that the current examples fulfill the requirement without going into too many details.

Reply:
Alright, I'll keep it at that for now and we can figure out use case based docs as we go 👍